file.c

```diff
@@ -7431,93 +7431,395 @@ Init_File(void)
+     *
+     * ==== File::NOATIME.
+     *
+     * Flag File::NOATIME specifies that act of opening the stream
+     * should not modify its access time (atime).
+     *
+     * Defined only for POSIX-compliant systems.
+     *
+     * ==== File::NOFOLLOW.
+     *
+     * Flag File::NOFOLLOW specifies that if path is a symbolic link,
+     * it should not be followed.
+     *
+     * Defined only for POSIX-compliant systems.
+     *
+     * ==== File::TMPFILE.
+     *
+     * Flag File::TMPFILE specifies that the opened stream
+     * should be a new temporary file.
+     *
+     * Defined only for POSIX-compliant systems.
+     *
+     * === Other File-Access \Constants
+     *
+     * ==== File::NONBLOCK
+     *
+     * When opening a FIFO with flags File::RDONLY or File::WRONLY:
+     *
+     * - If File::NONBLOCK is specified,
+     *   a read-only opened stream returns without delay.
+     *   A write-only opened stream returns an error
+     *   if no process currently has the file open for reading.
+     *
+     * - If File::NONBLOCK is not specified,
+     *   read-only opened stream blocks the calling thread
+     *   until a thread opens the file for writing.
+     *   A write-only stream blocks the calling thread until a thread opens the file
+     *   for reading.
+     *
+     * When opening a block special or character special file
+     * that supports non-blocking opens:
+     *
+     * - If File::NONBLOCK is specified, the opened stream returns without blocking
+     *   for the device to be ready or available.
+     *   Subsequent behavior of the device is device-specific.
+     *
+     * - If File::NONBLOCK is not specified, opened stream blocks the calling thread
+     *   until the device is ready or available before returning.
+     *
+     * - Otherwise, the behavior of File::NONBLOCK is unspecified.
+     *
+     * If path refers to a {STREAMS device}[https://en.wikipedia.org/wiki/STREAMS],
+     * the flags may be constructed from File::NONBLOCK
+     * along with File::RDONLYj, File::WRONLY, or File::RDWR.
+     * Other flags values have no effect on STREAMS devices.
+     *
+     * For a STREAMS device, the implementation of File::NONBLOCK is device-specific.
+     *
+     * ==== File::BINARY
+     *
+     * Flag File::BINARY specifies that the stream is to be accessed in binary mode;
+     * see {Data Mode}[rdoc-ref:File@Data+Mode].
+     *
+     * ==== File::NULL
+     *
+     * Flag File::NULL contains the string value of the null device:
+     *
+     * - On a Unix-like OS, <tt>'/dev/null'</tt>.
+     * - On Windows, <tt>'NUL'</tt>.
+     *
+     * ==== File::SHARE_DELETE
+     *
+     * Flag File::SHARE_DELETE enables other processes to open the stream
+     * with delete access.
+     *
+     * If the stream is opened for (local) delete access without File::SHARE_DELETE,
+     * and another process attempts to open it with delete access,
+     * the attempt fails and the stream is not opened for that process.
```

Comment:
We should make it clear that this is a Windows only feature.

Reply:
Fixed.